yt/frontends/enzo_e/fields.py

```diff
@@ -143,17 +143,34 @@ def setup_particle_mass_field(self, ptype):
                 constants = (constants,)
             names = [c[0] for c in constants]
 
+        group_list = nested_dict_get(
+            self.ds.parameters, ("Particle", ptype, "group_list"), default=()
+        )
+
         if "mass" in names:
-            val = constants[names.index("mass")][2]
-            val = self.ds.quan(val, self.ds.mass_unit)
-            if self.ds.cosmological_simulation:
+            # Prior to the addition of the is_gravitating group list,
+            # particle masses were stored as densities if the simulation
+            # was cosmological.
```

Comment:
This is incorrect. Particle "masses" were always treated as densities, independent of whether the simulation was cosmological or not.

Reply:
Thanks for clarifying this. This misconception dates back to the creation of this frontend, possibly when things were different in Enzo-E, or maybe just what I was able to glean from the data I had on hand. I will correct this.